pasqal_cloud/__init__.py

```diff
@@ -251,28 +254,108 @@ def rebatch(
             job_ids: list of job ids to retry.
```

Comment:
We should clarify the behaviour of filters in this MR stating that they are multiplicative ('AND').

For each item give more details about how it behaves. 
For example il the user passes a list of statuses, we will rebatch all the jobs whose status is found in the list.